src/arch/x86/interface/pcbios/e820mangler.S

```diff
@@ -24,6 +24,7 @@
 FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL )
 
 	.text
+	.section ".note.GNU-stack", "", @progbits
```

Comment:
This should be before the `.text` declaration.  (`.text` sets the current section name, so putting `.section "..."` after `.text` does change the destination section for the following statements.)

Reply:
Fixed in https://github.com/ipxe/ipxe/pull/899/commits/ddc27dc16bd8e1afa6d03b1c01ee6fed07248b65